pyre/pyre-interpreter/src/module/_multibytecodec/app_multibytecodec.py

```
@@ -25,18 +80,46 @@ def decode(self, object, final=False):
     def getstate(self):
-        return self.pending, 0
+        return self.pending, self.state
 
     def setstate(self, state):
-        self.pending = state[0]
+        # `mbidecoder_setstate` parses `(bytes, int)`; the pending buffer holds
+        # at most MAXDECPENDING bytes.
+        if not isinstance(state, tuple):
+            raise TypeError(
+                f"setstate() argument must be tuple, not {type(state).__name__}"
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)